### ISSUE TITLE
[Phase 1] Implement Core National-Municipal Coupling Model

### ISSUE THREAD

**Issue by bernardocaldas** [estimadorpt/models#38]
## Objective
Build the foundational model that learns how strongly each municipality (308 total) couples to national political trends versus following its own historical patterns.

## Data Requirements

**National political signal**: Extract `national_trend_pt` (pre-softmax latent values) from existing parliamentary model for election day

**Historical municipal results**: Load elections from 2009, 2013, 2017, 2021 (CNE data)
- Parties: PS, PSD, CDU, CDS-PP, BE, CH, IL, OTHER
- For coalitions (e.g., PSD.CDS-PP), sum component party support

**Municipality metadata**: Basic identification (code, name, district)

### Calculate historical baseline for each municipality-party:
- Average vote shares across training elections
- Add small constant (0.001) to avoid log(0)
- Convert to centered log-ratio: `log(p_party) - mean(log(all_parties))`

### Define coupling parameter:
- One beta per municipality
- Prior: Beta(2, 2) - uniform prior since we're not using urban/rural yet
- Represents weight given to national signal (0 = pure local, 1 = pure national)

### Combine signals on logit scale:
```
Linear_predictor[m,p] = beta[m] × National_logit[p] + (1-beta[m]) × Historical_logit[m,p]
```

## Model Structure
```python
with pm.Model():
    # Coupling strength per municipality
    coupling = pm.Beta('coupling', alpha=2, beta=2, dims='municipalities')
    
    # Combine national and local signals
    linear_pred = coupling × national_logit + (1-coupling) × historical_logit
    
    # Vote shares
    vote_shares = softmax(linear_pred)
    
    # Likelihood for observed results
    pm.DirichletMultinomial('results', n=total_votes, a=concentration*vote_shares, observed=observed)
```

## Validation
- Train on 2009, 2013, 2017
- Test on 2021
- Primary metric: Winner prediction accuracy
- Secondary: Vote share MAE
- Target: 60-65% winner accuracy

## Deliverables
- [ ] Data loading pipeline from CNE files
- [ ] Historical baseline calculation
- [ ] PyMC model implementation
- [ ] Training script
- [ ] Validation notebook showing 2021 predictions
- [ ] Analysis of coupling parameters by municipality
